app/lib/features/chat/providers/notifiers/chat_room_notifier.dart

```diff
@@ -926,20 +989,23 @@ class ChatRoomNotifier extends StateNotifier<ChatRoomState> {
     }
   }
 
-  ProfileData? getUserProfile(String userId) {
+  ProfileData? getUserProfile(String? userId) {
```

Comment:
why do we want to support getting a null here? Why would this ever be called with `null`? I think we should keep it as `String userId` and handle that non existence elsewhere.. 

It's more clear if there is a `userId != null ?  getUserProfile(userId) : null` that we know and handle `userid` to be `null` rather than wondering why a call of `getUserProfiel(userId)` doesn't return for the case (because the user is `null`)...

```suggestion
  ProfileData? getUserProfile(String userId) {
```